src/main/java/com/minecolonies/coremod/client/gui/DialogDoneCancel.java

```diff
@@ -0,0 +1,192 @@
+     * Constructor for the DialogDoneCancel class.
+     *
+     * @param window in which the dialog will be displayed
+     */
+    public DialogDoneCancel(final Window window)
+    {
+        super();
+        Loader.createFromXMLFile(Constants.MOD_ID + DIALOG_OK_CANCEL_SUFFIX, this);
+        titleLabel = findPaneOfTypeByID("title", Label.class);
+        contentText = findPaneOfTypeByID("textcontent", Text.class);
+        doneButton = findPaneOfTypeByID("done", Button.class);
+        cancelButton = findPaneOfTypeByID("cancel", Button.class);
+        doneButton.setHandler(this);
+        cancelButton.setHandler(this);
+        this.window = window;
+        setPosition(0, 0);
+        setVisible(false);
+    }
+
+    /**
+     * Set the title of the dialog.
+     *
+     * @param title for the dialog
+     */
+    public void setTitle(final String title)
+    {
+        titleLabel.setLabelText(title);
+    }
+
+    /**
+     * Get the title of the dialog.
+     *
+     * @return title for the dialog
+     */
+    public String getTitle()
+    {
+        return titleLabel.getLabelText();
+    }
+
+    /**
+     * Set the textual content of the dialog.
+     *
+     * @param content to display in the dialog
+     */
+    public void setTextContent(final String content)
+    {
+        contentText.setTextContent(content);
+    }
+
+    /**
+     * Get the textual content of the dialog.
+     *
+     * @return The textual content displayed in the dialog
+     */
+    public String getTextContent()
+    {
+        return contentText.getTextContent();
+    }
+
+    /**
+     * When a button have been cicked on.
+     *
+     * @param button which have been clicked on.
+     */
+    public void onButtonClicked(Button button)
+    {
+        this.setVisible(false);
+        if (handler == null)
+        {
+            Log.getLogger().error("DialogDoneCancel does not have a handler.");
+            return;
+        }
+        if (button == doneButton)
+        {
+            handler.onDialogClosed(this, DONE);
+        }
+        else
+        {
+            handler.onDialogClosed(this, CANCEL);
+        }
+    }
+
+    @Override
+    public void setVisible(final boolean visible)
+    {
+        if (visible)
+        {
+            setSize(window.getInteriorWidth(), window.getInteriorHeight());
+            //Make sure we are on top
+            putInside(window);
+        }
+        super.setVisible(visible);
+    }
+
+    /**
+     * Open the dialog.
+     */
+    public void open()
+    {
+        setVisible(true);
+    }
+
+    /**
+     * Close the dialog.
+     */
+    public void close()
+    {
+        setVisible(true);
+    }
+
+    /**
+     * Set the button handler for this button.
+     *
+     * @param h The new handler.
+     */
+    public void setHandler(final Handler h)
+    {
+        handler = h;
+    }
+
+    /**
+     * Used for windows that have dialog message.
+     */
+    @FunctionalInterface
+    public interface Handler
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Drop this interface in favor of "java.util.function.ObjIntConsumer<DialogDoneCancel>". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1711)